src/test/article.spec.js

```diff
@@ -304,3 +311,246 @@ describe('GET api/v1/articles', () => {
+      await generateArticle({ authorId: author.id })
+    );
+    const createdReport = await reportArticle(
+      createdArticle.id,
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination',
+        description: 'description'
+      })
+    );
+    const { violation, description } = createdReport[0].dataValues;
+
+    const token = helper.jwtSigner({ id: reporter.id });
+    const res = await server()
+      .post(`/api/v1/articles/sometest/report`)
+      .set('token', token)
+      .send({ violation, description });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal(
+      'Article Id must be a positive integer greater than zero'
+    );
+  });
+
+  it('should fail if fake token is  provided', async () => {
+    const reporter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+
+    const createdArticle2 = await createArticle(
+      { id: 2000 },
+      await generateArticle({ authorId: author.id })
+    );
+    const createdReport = await reportArticle(
+      createdArticle.id,
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination',
+        description: 'description'
+      })
+    );
+    const { violation, description } = createdReport[0].dataValues;
+
+    const fakeToken = 'asdfghu8765432';
+    const res = await server()
+      .post(`/api/v1/articles/${createdArticle2.id}/report`)
+      .set('token', fakeToken)
+      .send({ violation, description });
+    expect(res.status).to.equal(401);
+    expect(res.body.message).to.equal('Token is not valid');
+  });
+
+  it('should return validation error if violation is  not provided', async () => {
+    const reporter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+
+    const createdArticle2 = await createArticle(
+      { id: 2000 },
+      await generateArticle({ authorId: author.id })
+    );
+    const createdReport = await reportArticle(
+      createdArticle.id,
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination',
+        description: 'description'
+      })
+    );
+    const { description } = createdReport[0].dataValues;
+    const token = helper.jwtSigner({ id: reporter.id });
+    const res = await server()
+      .post(`/api/v1/articles/${createdArticle2.id}/report`)
+      .set('token', token)
+      .send({ description });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal('violation is required');
+  });
+
+  it('should return validation error if description is  not provided', async () => {
```

Comment:
This test should only be for cases where the `violation` === 'others'.

That's the only instance where a description is required